### DISCUSSION TITLE
Get default tenant name even when using Shellscope

### DISCUSSION THREAD

**Discussion by msquare98** [OrchardCMS/OrchardCore#11047] [Q&A] [answered]
I have a 'Default' tenant and multiple tenants to use.  If i understand it correctly, the tenant i am logged will automatically become the current tenant. I am using Shellscope to do something on my other tenants (Example: I am scoping to 'Tenant1' from 'Default' to do something). My question is, even though i am scoping to Tenant1, my original logged in Tenant is 'Default'. I want the name of original tenant i.e 'Default' even though i am scoped on to 'Tenant1'. Is it possible for me to get the  name of original logged in Tenant?

**Comment by Piedone** [comment, marked as answer]
Do you mean that when calling `shellScope.UsingAsync()` you'd like to see the "parent" shell's name in the executed delegate? I don't think there's a built-in way to do that. However, you can just pass the current shell's name (from an injected `ShellSettings`) to the delegate as a captured variable. Then, you can store it e.g. in a scoped service of yours that you can access deeper in the call chain and read out the value.

**Comment by ns8482e** [comment]
shellScope is for current request and if you want to access other tenant scope you can create and perform operation if you know host name

see

https://github.com/OrchardCMS/OrchardCore/discussions/9943

 I had similar situation but for performing some action from host to tenants
